web_app_drawer/static/src/js/web_app_drawer.js

```diff
@@ -0,0 +1,232 @@
+            if (this.isMobile()) {
+                event.preventDefault();
+            }
+        },
+        
+        // For lack of Modernizr, TouchEvent will do
+        isMobile: function () {
+            try{
+                document.createEvent('TouchEvent');
+                return true;
+            } catch (ex) {
+                return false;
+            }
+        },
+        
+    });
+    
+    var AppDrawer = Class.extend({
+        
+        LEFT: 'left',
+        RIGHT: 'right',
+        UP: 'up',
+        DOWN: 'down',
+        
+        isOpen: false,
+        $selectedAppLink: $(false),
+        keyBuffer: '',
+        keyBufferTime: 500,
+        keyBufferTimeoutEvent: false,
+        
+        init: function() {
+            this.$el = $('.drawer');
+            this.$el.drawer();
+            this.directionCodes = {
+                37: this.LEFT,  // Left arrow
+                39: this.RIGHT,  // Right arrow
+                38: this.UP,  // Up arrow
+                33: this.UP,  // page up
+                40: this.DOWN,  // Down arrow
+                34: this.DOWN,  // page down
+            };
+            var $clickZones = $('.openerp_webclient_container, ' +
+                                'a.oe_menu_leaf, ' +
+                                'a.oe_menu_toggler'
+                                );
+            $clickZones.click($.proxy(this.handleClickZones, this));
+            this.$el.one('drawer.opened', $.proxy(this.onDrawerOpen, this));
+            // Core bus keypress doesn't seem to send arrow keys
+            core.bus.on('keydown', this, this.handleNavKeys);
+        },
+        
+        // It provides handlers to hide drawer when "unfocused"
+        handleClickZones: function() {
+            this.$el.drawer('close');
+            $('.oe_secondary_menus_container')
+                .parent()
+                .collapse('hide');
+        },
+        
+        // It provides keyboard shortcuts for app drawer nav
+        handleNavKeys: function(e) {
+            if (!this.isOpen){
+                return;
+            }
+            if (Object.keys(this.directionCodes).indexOf(e.keyCode.toString()) !== -1) {
+                var $link = this.findAdjacentAppLink(
+                    this.$selectedAppLink,
+                    this.directionCodes[e.keyCode]
+                );
+                this.selectAppLink($link);
+            } else {
+                var buffer = this.handleKeyBuffer(e.keyCode);
+                this.selectAppLink(this.searchAppLinks(buffer));
+            }
+        },
+        
+        // It adds to keybuffer, sets expire timer, and returns buffer
+        handleKeyBuffer: function(keyCode) {
+            this.keyBuffer += String.fromCharCode(keyCode);
+            if (this.keyBufferTimeoutEvent) {
+                clearTimeout(this.keyBufferTimeoutEvent);
+            }
+            this.keyBufferTimeoutEvent = setTimeout(this.clearKeyBuffer, this.keyBufferTime);
```

Comment:
Again, you need `$.proxy(this.clearKeyBuffer, this)` if you don't want that `this` becomes the `Window` object.

![captura de pantalla de 2016-08-25 10-04-15](https://cloud.githubusercontent.com/assets/973709/17961326/55140854-6aab-11e6-8d0b-b1dfdc934c7a.png)